src/rabbit_delayed_message.erl

```diff
@@ -50,28 +48,12 @@
                              delay()) ->
                                     nodelay | {ok, t_reference()}.
 
--define(TABLE_NAME, append_to_atom(?MODULE, node())).
--define(INDEX_TABLE_NAME, append_to_atom(?TABLE_NAME, "_index")).
+-define(Timeout, 5000).
```

Comment:
wonder if timeout should be 1 second as that is the resolution of the `x-delay` parameter. (Maybe some users would be surprised/unhappy if their 1 second delay is delivered after 5 seconds instead)
The previous solution of keeping track of the closest timestamp was also clever. Do you think it added too much complexity? (It definitely makes it necessary to keep the writer and reader in the same process)

Reply:
The timeout can be configurable. It does not add complexity (other than an upper limit), but is a different design compared to the one in the design doc.